src/serlo-editor/plugins/fill-in-the-gap-exercise/renderer.tsx

```diff
@@ -0,0 +1,216 @@
+                />
+              ))}
+          </>
+        </GapSolutionsArea>
+      ) : null} */}
+
+      {/* Copied from mc-renderer.tsx */}
+      <div className="mt-2 flex">
+        <button
+          className="serlo-button-blue mr-3 h-8"
+          onClick={() => {
+            checkAnswers()
+            setShowFeedback(true)
+          }}
+        >
+          {exStrings.check}
+        </button>
+        {showFeedback && (
+          <Feedback
+            correct={[...gapFeedback].every((entry) => entry[1].isCorrect)}
+          />
+        )}
+      </div>
+
+      {/* Only debug output from here on */}
+      <div className="hidden">
+        Gaps state:
+        {gapStateList.map((gap, index) => (
+          <div key={index}>{JSON.stringify(gap)}</div>
+        ))}
+      </div>
+      <div className="hidden">
+        <div>State textUserTypedIntoGap:</div>
+        {[...textUserTypedIntoGap].map((entry, index) => {
+          const gapId = entry[0]
+          const text = entry[1].text
+          return (
+            <div
+              className="ml-5"
+              key={index}
+            >{`Text: ${text} | GapId: ${gapId}`}</div>
+          )
+        })}
+      </div>
+      {/* <div className="hidden">
+        {gapDragAndDropSolutions.map((entry, index) => (
+          <div
+            key={index}
+          >{`DraggableId: ${entry.draggableId} | in droppableId: ${entry.inDroppableId} | containing text: ${entry.text}`}</div>
+        ))}
+      </div> */}
+    </div>
+  )
+
+  function checkAnswers() {
+    if (mode === 'fill-in-the-gap') {
+      const newGapAnswersCorrectList = new Map<
+        GapId,
+        { isCorrect: boolean | undefined }
+      >()
+      gapStateList.forEach((gapState) => {
+        const textUserTypedIntoThisGap =
+          textUserTypedIntoGap.get(gapState.gapId)?.text ?? ''
+        const answerCorrect =
+          textUserTypedIntoThisGap === gapState.correctAnswer ||
+          textUserTypedIntoThisGap ===
+            gapState.alternativeSolutions.find(
+              (alternativeSolution) =>
+                textUserTypedIntoThisGap === alternativeSolution
+            )
+        newGapAnswersCorrectList.set(gapState.gapId, {
+          isCorrect: answerCorrect,
+        })
+      })
+
+      setGapFeedback(newGapAnswersCorrectList)
+    } else if (mode === 'drag-and-drop') {
+      // TODO: Check answers in drag-and-drop mode
+    }
+  }
+
+  // Searches for gap objects in text plugin state. They can be at varying depths.
+  function getGapsWithinObject(obj: object) {
+    if (GapState.is(obj)) {
+      return [obj]
+    }
+
+    // Recursively search this objects values for gap objects
+    let objList: t.TypeOf<typeof GapState>[] = []
+    Object.values(obj).forEach((_value) => {
+      const value: unknown = _value
+      if (typeof value === 'object' && value !== null) {
+        objList = [...objList, ...getGapsWithinObject(value)]
+      }
+    })
+
+    return objList
```

Comment:
I thought that using reduce would reduce (ba-dum-reducce) the amount of code here. In the end, it kind of didn't 😄 Still, here's the code, take it or leave it
```suggestion
    return Object.values(pluginState).reduce(
      (gaps: t.TypeOf<typeof GapState>[], value: unknown) => {
        if (typeof value === 'object' && value !== null) {
          gaps.concat(getGapsWithinObject(value))
        }
        return gaps
      },
      []
    )
```
(btw, renamed `obj` to `pluginState` for clarity)

Reply:
I like it! Thank you for the suggestion.